From: Tetsu Yamamoto <tyamamot@redhat.com>
Date: Thu, 24 Jan 2008 11:35:29 -0500
Subject: [xen] ia64: stop all cpus on hv panic part2
Message-id: 4798BE51.3030000@redhat.com
O-Subject: Re: [RHEL5.2 PATCH] [Xen] Dom0/DomU may continue to work after hypervisor panic
Bugzilla: 426129

Hi,

The patch for BZ#426129 was included in kernel -69,
xen-ia64-stop-all-cpus-on-hv-panic.patch, but it seems to be lacking one
part of the posted patch.  Could you please check it and add the
following part?

diff --git a/arch/ia64/xen/domain.c b/arch/ia64/xen/domain.c
index b03e265..f01b027 100644
--- a/arch/ia64/xen/domain.c
+++ b/arch/ia64/xen/domain.c
@@ -1351,10 +1351,12 @@ extern void cpu_halt(void);
 void machine_halt(void)
 {
 	console_start_sync();
-	if (running_on_sim)
-		printk ("machine_halt called.  spinning...\n");
-	else
-		cpu_halt();
+
+#ifdef CONFIG_SMP
+	smp_send_stop();
+#endif
+
+	printk ("machine_halt called.  spinning...\n");
 	while(1);
 }
 
